app/models/lti_deployment.rb

```diff
@@ -2,4 +2,29 @@ class LtiDeployment < ApplicationRecord
   belongs_to :course, optional: true
   belongs_to :lti_client
   validates :external_deployment_id, uniqueness: { scope: :lti_client }
+
+  def get_students
+    auth_data = lti_client.get_oauth_token(['https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly'])
+    names_service = LtiService.find_by!(lti_deployment: self, service_type: 'namesroles')
+    membership_uri = URI(names_service.url)
+    membership_uri.query = URI.encode_www_form(role: 'http://purl.imsglobal.org/vocab/lis/v2/membership#Learner')
+    req = Net::HTTP::Get.new(membership_uri)
+    req['Authorization'] = "#{auth_data['token_type']} #{auth_data['access_token']}"
+    http = Net::HTTP.new(membership_uri.host, membership_uri.port)
+
+    res = http.request(req)
+    member_info = JSON.parse(res.body)
+    member_info['members'].each do |member|
```

Comment:
Okay so overall this loop seems like it's going to be pretty slow when the classes get large (it's a form of N+1 querying). We have an opportunity to use some bulk ActiveRecord operations to speed things up. (I don't think you've done much of this, but it's a good experience.)

Check out https://edgeapi.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-upsert_all, and in particular note that you can get the method to return the `id`s of the inserted/updated records.